nilearn/_utils/data_gen.py

```diff
@@ -751,6 +750,25 @@ def basic_confounds(length, random_state=0):
     return confounds
 
 
+def add_metadata_to_bids_derivatives(bids_path: str | Path,
+                                     metadata: dict,
+                                     json_file: str = None) -> Path:
```

Comment:
I thought we were holding off for now

Reply:
you are correct
I had added them before we made a decision: I thought I had removed all of them but I missed a fews